### ISSUE TITLE
✅ 🐞 Parsing null to parameter #1 ($string) in parse_str() causes WooCommerce Checkout to not work if PHP8.1+ is in use - [ED-13773]

### ISSUE THREAD

**Issue by DeveloperWil** [elementor/elementor#24880]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

Running Elementor, Elementor Pro and WooCommerce on PHP 8.1.  
The checkout fields are greyed out and I get the circle ajax spinner indefinitely.
The following error core is in the debug.log file.
[22-Jan-2024 23:59:20 UTC] PHP Deprecated:  parse_str(): Passing null to parameter #1 ($string) of type string is deprecated in /home/zwxhxqss/public_html/wp-content/plugins/elementor-pro/modules/woocommerce/module.php on line 626
[22-Jan-2024 23:59:20 UTC] PHP Deprecated:  parse_str(): Passing null to parameter #1 ($string) of type string is deprecated in /home/zwxhxqss/public_html/wp-content/plugins/elementor-pro/modules/woocommerce/module.php on line 626

If I revert back to PHP 8.0 or 7.4, the issue goes away.

It seems there is a null parameter being passed to parse_str() in the code, which PHP 8.1 does not like because of its strong type detection.

### Steps to reproduce

Install new version of WordPress.  Turn on debugging and file log.
Install Elementor, Elementor Pro and WooCommerce.
Go through WooCommerce setup wizard.
Publish a a test product.
Set PHP version to 8.1.
Add product to cart and visit checkout.
Checkout fields are greyed out with ajax spinner indefinitely.
Check wp-debug.log

### Expected behavior

I'd expect the checkout fields to be accessible and not endlessly greyed out with the Ajax circle spinner.

### Isolating the problem

- [X] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [ ] This bug happens with the Hello Elementor theme active.
- [X] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

````txt

== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MySQL Community Server - (GPL) v5.7.44
	PHP Version: 8.0.30
	PHP Memory Limit: 256M
	PHP Max Input Vars: 1000
	PHP Max Post Size: 64M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.4.2
	Site URL: https://wosg.com.au
	Home URL: https://wosg.com.au
	WP Multisite: No
	Max Upload Size: 64 MB
	Memory limit: 40M
	Max Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: en-AU
	Timezone: Australia/Sydney
	Debug Mode: Active

== Theme ==
	Name: Twenty Twenty-Four
	Version: 1.0
	Author: the WordPress team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_AU
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36

== Active Plugins ==
	Elementor
		Version: 3.18.3
		Author: Elementor.com

	Elementor Pro
		Version: 3.18.3
		Author: Elementor.com

	Temporary Login Without Password
		Version: 1.8.3
		Author: StoreApps

	WooCommerce
		Version: 8.5.1
		Author: Automattic


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	
	woocommerce: Active


== Elementor Experiments ==
	Optimised DOM Output: Active by default
	Improved Asset Loading: Active by default
	Improved CSS Loading: Inactive by default
	Inline Font Icons: Inactive
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Inactive by default
	Grid Container: Active
	Default to New Theme Builder: Active by default
	Editor Top Bar: Active
	Optimized Gutenberg Loading: Active
	Build with AI: Inactive
	Landing Pages: Active by default
	Nested Elements: Inactive by default
	Lazy Load Background Images: Inactive
	Optimize Image Loading: Active by default
	Pages Panel: Inactive by default
	Global Style Guide: Active by default
	Page Transitions: Active by default
	Notes: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default
	Menu: Inactive by default
	Taxonomy Filter: Inactive by default


== Log ==
	
Log: showing 20 of 362023-12-09 12:25:44 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.18.0',
  'to' => '3.18.2',
)]
2023-12-09 12:25:44 [info] elementor-pro::elementor_pro_updater Started 
2023-12-09 12:25:44 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-12-09 12:25:44 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-12-09 12:25:44 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.17.1',
  'to' => '3.18.1',
)]
2023-12-27 12:04:59 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.18.2',
  'to' => '3.18.3',
)]
2023-12-27 12:05:12 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.18.1',
  'to' => '3.18.2',
)]
2023-12-27 13:17:33 [info] elementor::elementor_updater Started 
2023-12-27 13:17:33 [info] Elementor/Upgrades - _on_each_version Start  
2023-12-27 13:17:33 [info] Elementor/Upgrades - _on_each_version Finished 
2023-12-27 13:17:33 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.18.2',
  'to' => '3.18.3',
)]
2023-12-27 13:17:33 [info] elementor-pro::elementor_pro_updater Started 
2023-12-27 13:17:33 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-12-27 13:17:33 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-12-27 13:17:33 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.18.1',
  'to' => '3.18.2',
)]
2024-01-20 10:15:14 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.18.2',
  'to' => '3.18.3',
)]
2024-01-20 11:08:40 [info] elementor-pro::elementor_pro_updater Started 
2024-01-20 11:08:40 [info] Elementor Pro/Upgrades - _on_each_version Start  
2024-01-20 11:08:40 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2024-01-20 11:08:40 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.18.2',
  'to' => '3.18.3',
)]

PHP: showing 4 of 4PHP: 2023-11-10 10:47:06 [notice X 1][../wp-content/plugins/elementor/includes/elements/container.php::43] Creation of dynamic property Elementor\Includes\Elements\Container::$logical_dimensions_inline_end is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-11-26 14:03:49 [notice X 70][../wp-content/plugins/elementor-pro/modules/woocommerce/module.php::607] parse_str(): Passing null to parameter #1 ($string) of type string is deprecated [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-12-06 20:13:20 [warning X 78][../wp-content/plugins/elementor-pro/modules/theme-builder/widgets/site-logo.php::192] Undefined array key &quot;options&quot; [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2024-01-22 18:57:54 [warning X 1][../wp-content/plugins/elementor/modules/history/revisions-manager.php::363] Undefined array key &quot;editor_post_id&quot; [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 2 of 2JS: 2024-01-21 05:46:53 [error X 17][../wp-content/uploads/essential-addons-elementor/eael.js?ver=1614335014:28:480733] Cannot read properties of undefined (reading &#039;size&#039;) 
JS: 2024-01-22 07:58:28 [error X 1][../wp-admin/load-scripts.php?c=0&amp;loadchunk_0=jquery-core,jquery-migrate,utils,moxiejs,plupload&amp;ver=6.4.2:2:28760] Cannot read properties of undefined (reading &#039;value&#039;) 



== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible

== Elementor Pro - Compatibility Tag ==

````

</details>

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by freschstudio** [comment]
I just stumbled on this error on a production website. I'm using the latest Elementor Pro, Tutor LMS and PHP 8.2.

What fixed it for me was disabling WP_DEBUG `define( 'WP_DEBUG', false );` in wp-config.php. It was accidentally left on. 

Note: This is more of a quick workround that a proper fix.

**Comment by rrivera-ixis** [comment]
This is happening because there's a problem with the initial AJAX call when the page loads `?wc-ajax=update_order_review`. A deprecation warning that messes up with the response:

`Deprecated: parse_str(): Passing null to parameter #1 ($string) of type string is deprecated in /public_html/wp-content/plugins/elementor-pro/modules/woocommerce/module.php on line 626`

Disabling WP_DEBUG removes this message from the request resulting in the expected output.

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro v3.20.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
